pg_shard.c

```diff
@@ -1004,6 +1402,135 @@ TargetEntryList(List *expressionList)
 
 
 /*
+ * TargetEntryVarList get a target list and creates a new target list entry
+ * whose elements are Vars, which are pulled from the input target list.
+ */
+static List *
+TargetEntryVarList(List *targetEntryList)
```

Comment:
This code is very similar to `flatten_tlist`, provided by PostgreSQL. Is there any way we could use that instead?